gpu_ocean/SWESimulators/CDKLM16.py

```diff
@@ -275,6 +296,45 @@ def __init__(self, \
         self.angle_texref.set_address_mode(1, cuda.address_mode.CLAMP)
         self.angle_texref.set_flags(cuda.TRSF_NORMALIZED_COORDINATES) #Use [0, 1] indexing
         
+        
+        
+        # Texture for coriolis f
+        self.coriolis_texref = self.kernel.get_texref("coriolis_f_tex")
+        
+        # Create the CPU coriolis
+        if (latitude is not None):
+            if (self.f != 0.0):
+                raise RuntimeError("Cannot specify both latitude and f. Make your mind up.")
+            if (subsample_latitude and latitude.shape == eta0.shape):
+                self.logger.info("Subsampling latitude texture by factor " + str(subsample_latitude))
+                latitude = subsample(latitude, subsample_latitude)
+            coriolis_f, _ = OceanographicUtilities.calcCoriolisParams(latitude)
+            coriolis_f = coriolis_f.astype(np.float32)
+        else:
+            if (self.coriolis_beta != 0.0):
+                if (angle.size != 1):
+                    raise RuntimeError("non-constant angle cannot be combined with beta plane model (makes no sense)")
+                x = np.linspace(0, self.nx*self.dx, self.nx//subsample_f)
+                y = np.linspace(0, self.ny*self.dy, self.ny//subsample_f)
+                self.logger.info("Using latitude to create Coriolis f texture ({:f}x{:f} cells)".format(x.size, y.size))
+                x, y = np.meshgrid(x, y)
+                n = x*np.sin(angle[0, 0]) + y*np.cos(angle[0, 0])
+                coriolis_f = self.f + self.coriolis_beta*(n)
+            else:
+                if (self.f.size != 1):
+                    raise NotImplementedError("(User specified) varying f is not implemented yet, but is trivial to implement. Use varying lat instead")
```

Comment:
Honestly, I think trivial is a bit strong.

It took  me some time to get this suggestion. 
```
def initLatitude(lat):
    ny, nx = lat.shape
    
    #North pole
    x0 = 0.5
    y0 = 2.0
    
    #Distance to north pole
    x = (np.linspace(-1, 1, nx) - x0)*width/2
    y = (np.linspace(-1, 1, ny) - y0)*height/2
    x, y = np.meshgrid(x, y)
    
    # The distance between latitudes are 111.2 km
    # Numbers of latitudes from the north pole
    l = np.sqrt(x**2 + y**2)/111200
    
    lat = 90 - l
    return lat
```

It looks fine with the angle from the convergence notebook plotted like this:
```
angle = np.zeros(dataShape, dtype=np.float32)
lat = np.zeros(dataShape, dtype=np.float32)
angle=initAngle(angle)
lat = initLatitude(lat)
plt.figure(figsize=(4,4))

plt.quiver(np.sin(angle[::32, ::32]), np.cos(angle[::32, ::32]), color='r')
plt.quiver(np.cos(angle[::32, ::32]), -np.sin(angle[::32, ::32]), color='g')
plt.contour(lat[::32, ::32])
plt.colorbar()
```

Reply:
User specified f - hvis du har et array f, så er det bare å sette
```
coriolis_f = np.array(self.f, dtype=np.float32)
``` 
siden CUDA-kjernen kun bryr seg om f, ikke lat/beta/f_0, etc. Eneste grunn til at jeg ikke gjorde dette var pga tekstur-koordinatene som ikke er 100% sjekket. Så jeg står ved "trivial".

Reply:
Forstod ikke helt kommentaren din her. Feilmeldingen sier jo at
```
coriolis_f = np.array(self.f, dtype=np.float32)
```
ikke støttes?

Reply:
Se oppdatert commit. Jeg har implementert det nå og kaster feil hvis den ikke matcher størrelse 1 eller størrelse == eta0.shape.

Reply:
Poenget var det vi snakket om i lunsjen i går - det med ghostceller etc, og at jeg ikke hadde tid til å sette meg ned å implementere / sjekke at det ble riktig. Har sjekket nå ref disse +/-2-sakene. Så nå skal det være i orden. Du kan nå enten
- sette konstant f - dette gir en 1x1-tekstur med spesifisert f
- sette konstant f med beta  - dette gir nx*ny-tekstur med spesifisert f+lat*beta
- sette konstant latitude - dette gir en 1x1-tekstur med spesifisert lat
- sette varierende latitude - dette gir typisk nx*ny-tekstur med spesifisert lat

- Hvis du forsøker sette både latitude og f får du feil (dette gir ikke mening)
- Hvis du bruker konstant f+beta og varierende angle få rdu feil (dette gir ikke mening)